src/view/RestaurantView.java

```diff
@@ -105,6 +104,12 @@ public void propertyChange(PropertyChangeEvent evt) {
         info.add(restaurantAddress);
         info.add(restaurantPhoneNumber);
         info.add(restaurantCategories);
+
+        if (state.getPreviousView().equals("view restaurants")) {
+            buttons.add(addToFavourite);
+        } else if (state.getPreviousView().equals("view favourites")) {
+            buttons.remove(addToFavourite);
+        }
```

Comment:
If your action here involves toggling the visibility of the `addToFavorites` button, you can utilize `addToFavorites.setVisible(false)` inside `actionPerformed` to hide it, and similarly to show it again.

Reply:
Thanks! This is a much better solution.